### DISCUSSION TITLE
[Design] : How to edit the Rocketbot Written in the html page ?

### DISCUSSION THREAD

**Discussion by the-rough-page** [apache/skywalking#8778] [Q&A] [answered]
Before compiling is there a js folder or CSS file where the class rocket bot is there?
or 
after deploying it where can I find the source HTML page where I change the rocket bot written on the top?
or I want to add my CSS on that rocket bot so if get the class I'll design it accordingly.

**Comment by Superskyyy** [comment, marked as answer]
@the-rough-page Hello! If I understand correctly, you are asking about the way to customize the RocketBotUI of SkyWalking right? Or are you referring to changing this specific logo?
![image](https://user-images.githubusercontent.com/26076517/161157016-d17f07e7-4e8c-4542-8fd5-a153df558229.png)
 

Looks like you got a whole team ready, could you elaborate on what kinds of specific changes that you would make? Note that [RocketBotUI](https://github.com/apache/skywalking-rocketbot-ui) as I linked is being deprecated as of SkyWalking V9, we have a new BoosterUI [here ](https://github.com/apache/skywalking-booster-ui) that will be released pretty soon. You might want to work on that instead. Both projects are written in Vuejs.

**Reply by the-rough-page** [reply]
yes, I want to change this only the rocket bot written below skywalking
wanna style it and add something rocket bot to be exact.
can you tell me where to find the source HTML and CSS files after installation under the apn bin ?

i installed it in Cent os (java version)
we're using [this](https://github.com/apache/skywalking)

**Reply by Superskyyy** [reply]
> yes, I want to change this only the rocket bot written below skywalking wanna style it and add something rocket bot to be exact. can you tell me where to find the source HTML and CSS files after installation under the apn bin ?
> 
> i installed it in Cent os (java version)

No, you definitely do not want to edit a modern compiled frontend like this, this is not a basic HTML, JS, CSS project, it is written and compiled in Vue3.  What you need to do is retrieve a copy of the source [project ](https://github.com/apache/skywalking-rocketbot-ui) and modify the source code then compile your own version of UI, finally substitute the web app JAR. 

In case you are really curious about how to do the thing... You need to extract the JAR to locate the compressed source code with class name `grey rocketbot`, it is hidden somewhere in the huge `.js` file that is not intended for human to read. Note this method is not reusable at all and just a hack. I cannot imagine any scenario where this is a good way of modifying the project rather than toying around with it.

**Reply by the-rough-page** [reply]
no, I just wanted to style it and in the Source code also i was not able to find that `.js` file where i can see the class `Grey Rocketbot ` before the compilation can you suggest the name of the `.js` or `jar`  i should try.
and I'm not toying with it i just want to make it like rocket bot v2. so please help if you can.

**Reply by Superskyyy** [reply]
https://github.com/apache/skywalking-rocketbot-ui/blob/master/src/views/components/common/rk-header.vue It is right here in the source code.

**Reply by Superskyyy** [reply]
Be careful don't get confused, there is no .js file before packaging. It is generated only after transpiling and optimization by the Vue CLI.
